app/src/main/java/com/github/h3lp3rs/h3lp/professional/VerificationActivity.kt

```diff
@@ -0,0 +1,140 @@
+
+        // Database representing the professional users authenticated
+        val db = Databases.databaseOf(Databases.PRO_USERS)
+
+        // Reference to the Firebase cloud storage
+        var storageRef = FirebaseStorage.getInstance().getReference("uploads")
+
+        // Type of the intent to launch when choosing an image from the device
+        private const val INTENT_TYPE = "image/*"
+        private const val UPLOAD_FAILURE_MSG = "Upload failed. Try again!"
+        var currentUserId = SignInActivity.userUid.toString()
+        var currentUserName = GoogleSignInAdapter.auth.currentUser?.displayName.toString()
+
+    }
+
+    // Layout components
+    private lateinit var chooseImgButton: Button
+    private lateinit var uploadButton: Button
+    private lateinit var fileName: EditText
+    private lateinit var img: ImageView
+    private lateinit var progressBar: ProgressBar
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_verification)
+
+        chooseImgButton = findViewById(R.id.button_choose_img)
+        uploadButton = findViewById(R.id.button_upload)
+        fileName = findViewById(R.id.edit_file_name)
+        img = findViewById(R.id.file)
+        progressBar = findViewById(R.id.progress_bar)
+
+        chooseImgButton.setOnClickListener {
+            openFileChooser()
+        }
+        uploadButton.setOnClickListener {
+            uploadFile()
+        }
+    }
+
+    /**
+     * Opens the image file chooser of the device inorder to select an image
+     */
+    private fun openFileChooser() {
+        val intent = Intent()
+        intent.type = INTENT_TYPE
+        intent.action = Intent.ACTION_GET_CONTENT
+        resultLauncher.launch(intent)
+    }
+
+    /**
+     * Handles the intent result launched by the image file selection
+     */
+    private val resultLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_OK && result.data != null && result.data!!.data != null) {
+                imgUri = result.data!!.data!!
+                // Display the selected image
+                Picasso.with(this).load(imgUri).into(img)
+            }
+        }
+
+    /**
+     * Retrieves the file extension
+     *
+     * @param uri Uri of the file
+     * @return the extension of the file
+     */
+    private fun getFileExtension(uri: Uri): String? {
+        return MimeTypeMap.getSingleton().getExtensionFromMimeType(contentResolver.getType(uri))
+    }
+
+    /**
+     * Uploads the selected file into the cloud storage as well as the stores the current user in the authenticated users
+     * database
+     */
+    private fun uploadFile() {
+        if (imgUri != null) {
+            // Creates a file in the cloud storage with the current time as name (Sufficient to avoid collision but can be redesigned in the future )
```

Comment:
what about concatenating the uid?

Reply:
This could even use the incrementAndGet from my PR to get a unique id (but no worries if you don't have the time since this would require quite a few changes and your solution is clearly viable at this point in time, this could possibly constitute a small task for a coming sprint)

Reply:
You're right! It would be more elegant to use the incremebtAndGet! Noted 👍🏻